controllers/nstemplateset/namespaces.go

```diff
@@ -175,7 +183,16 @@ func (r *namespacesManager) delete(logger logr.Logger, nsTmplSet *toolchainv1alp
 			if err := r.Client.Delete(context.TODO(), &ns); err != nil {
 				return false, r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusTerminatingFailed, err, "failed to delete user namespace '%s'", ns.Name)
 			}
-			return true, nil
+
+			if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: ns.Name}, &corev1.Namespace{}); err != nil {
+				if errors.IsNotFound(err) {
+					return true, nil // namespace was actually deleted and thus not found by get
+				}
+			}
+			// No error implies namespace was not deleted
+			v := validationError{}
+			err = &v
```

Comment:
You can just use `fmt.Errorf("Namespace deletion wasn't complete")`

Reply:
that makes sense, updated.